airflow-core/tests/unit/dag_processing/test_processor.py

```diff
@@ -581,6 +582,63 @@ def fake_collect_dags(self, *args, **kwargs):
     assert called is True
 
 
+def test_callback_processing_does_not_update_timestamps(session):
+    """Callback processing should not update last_finish_time to prevent stale DAG detection."""
+    stat = process_parse_results(
+        run_duration=1.0,
+        finish_time=timezone.utcnow(),
+        run_count=5,
+        bundle_name="test",
+        bundle_version=None,
+        parsing_result=None,
+        session=session,
+        is_callback_only=True,
+    )
+
+    assert stat.last_finish_time is None
+    assert stat.run_count == 5
+
+
+def test_normal_parsing_updates_timestamps(session):
+    """last_finish_time should be updated when parsing dag file."""
```

Comment:
```suggestion
    """last_finish_time should be updated when parsing a dag file."""
```

Reply:
Fixed in https://github.com/apache/airflow/pull/55698/commits/326c7f97735367b9083302cc89f66233553e1711